### DISCUSSION TITLE
Query Combo Box should make use of dataobjformatter attribute in typesearch definition.

### DISCUSSION THREAD

**Discussion by benanhalt** [specify/specify7#5544] [Feature Requests]
The typesearch definition
```xml
<typesearch tableid="4" name="Taxon" searchfield="fullName" displaycols="fullName, author" format="%s %s" uifieldformatter="" dataobjformatter="" system="true"/>
```
should include the author field in the search results display.

This would fix https://github.com/RBGE-Herbarium/Specify_functionality/issues/11

A workaround is to change the default table formatter to include the desired columns.

**Comment by mcruz-umich** [comment]
This link:  https://github.com/RBGE-Herbarium/Specify_functionality/issues/11
Needs revision in the post above... it is getting a 404.

**Comment by maxpatiiuk** [comment]
It's a link to a super private repository.
Even I don't have access to it 🕵️

**Comment by mcruz-umich** [comment]
Haha - cool!

**Comment by maxpatiiuk** [comment]
Specify 7 is currently using the default formatter for this, which offers way more flexibility than just a list of field names, thus this is potentially a wont-fix.

There shouldn't be considered performance difference between using a formatter and just returning multiple fields as formatters are just constructing SQL expressions

The only remaining thing is the ability to use a formatter other than the default one. Thus, I will repurpose this issue for that. This is blocked by #1959

**Comment by grantfitzsimmons** [comment]
`displaycols` are also ignored, rather it uses the default formatter instead.

**Comment by maxpatiiuk** [comment]
something like displaycols was also requested for the Collection Relationship plugin too (to display data in multiple columns, rather than concatenated together)

probably the way to implement that in a futher-proof and flexible way is to allow passing a list of formatters rather than just one formatter - and each formatter would be a separate column
